evolver/history.py

```diff
@@ -10,13 +10,41 @@ class Config(BaseConfig):
         pass
 
     @abstractmethod
-    def put(self, name, data):
+    def set(self, name, data):
         pass
 
     @abstractmethod
     def get(self, query):
         pass
 
+    @classmethod
+    def track_history(cls, name: str = None):
+        """ Decorator for convenient tracking of data history.
+            E.g., decorate ``Device.get()`` to automatically track the data returned by the device.
+
+            Args:
+                name str: The name of the history stream to add data to.
+        """
+        def decorator(func):
+            def wrapper(self, *args, **kwargs):
+                nonlocal name
+                ret = func(self, *args, **kwargs)
+
+                if name is None:
+                    name = getattr(self, "name")
+
+                # Track history.
+                cls.getHistory().set(name=name, data=ret)
+
+                return ret
+            return wrapper
+        return decorator
+
+    @classmethod
+    def getHistory(cls, name: str = None):
```

Comment:
@amitschang We have options here, what's your pref:
- make ``history.history`` a class attribute similarly to ``logging.Logger.manager``
- singleton
- some additional registration func
- something else?